torch/nested/_internal/ops.py

```diff
@@ -884,16 +884,91 @@ def split_with_sizes_default(func, *args, **kwargs):
     ]
 
 
+# TODO: Implement slice() instead and narrow() in terms of slice()
 @register_jagged_func(
-    torch.ops.aten.narrow.default, "self: jt, dim: any, start: any, length: any"
+    torch.ops.aten.narrow.default, "self: jt_all, dim: any, start: any, length: any"
 )
 def narrow(func, *args, **kwargs):
     _, new_kwargs = normalize_function(  # type: ignore[misc]
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
     )
     inp = new_kwargs.pop("input")
 
-    dim = _wrap_jagged_dim(inp.dim(), new_kwargs["dim"], inp._ragged_idx, "narrow")
+    dim, operating_on_batch = _wrap_jagged_dim(
+        inp.dim(), new_kwargs["dim"], inp._ragged_idx, "narrow", allow_batch_dim=True
+    )
+    if operating_on_batch:
+        # batch dim narrowing requires custom logic involving offsets
+        out_kwargs = extract_kwargs(inp)
+        start_val, length_val = new_kwargs["start"], new_kwargs["length"]
```

Comment:
Trying to think of weird edge cases (totally fine if normal narrow doesn't cover this either, but I guess we want parity?)

- What happens if length_val is negative? I didn't see any explicit check, and  [docs](https://pytorch.org/docs/stable/generated/torch.narrow.html ) say that it must "weakly positive" whatever that means 🤔.

- What happens if start_val is negative, but length_val > size of the tensor.

Reply:
good call, will add test cases for these

>  it must "weakly positive" whatever that means

my understanding is that this allows for length=0? idk for sure though